commit 03881c05b948c2eb0e8b4fa86eeee89298313021
Author: Rob Landley <rob@landley.net>
Date:   Sat May 20 15:03:29 2017 -0500

    x86/boot: Use CROSS_COMPILE prefix for readelf
    
    commit 3780578761921f094179c6289072a74b2228c602 upstream.
    
    The boot code Makefile contains a straight 'readelf' invocation. This
    causes build warnings in cross compile environments, when there is no
    unprefixed readelf accessible via $PATH.
    
    Add the missing $(CROSS_COMPILE) prefix.
    
    [ tglx: Rewrote changelog ]
    
    Fixes: 98f78525371b ("x86/boot: Refuse to build with data relocations")
    Signed-off-by: Rob Landley <rob@landley.net>
    Acked-by: Kees Cook <keescook@chromium.org>
    Cc: Jiri Kosina <jkosina@suse.cz>
    Cc: Paul Bolle <pebolle@tiscali.nl>
    Cc: "H.J. Lu" <hjl.tools@gmail.com>
    Link: http://lkml.kernel.org/r/ced18878-693a-9576-a024-113ef39a22c0@landley.net
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/x86/boot/compressed/Makefile b/arch/x86/boot/compressed/Makefile
index 34d9e15..4669b3a 100644
--- a/arch/x86/boot/compressed/Makefile
+++ b/arch/x86/boot/compressed/Makefile
@@ -94,7 +94,7 @@ vmlinux-objs-$(CONFIG_EFI_MIXED) += $(obj)/efi_thunk_$(BITS).o
 quiet_cmd_check_data_rel = DATAREL $@
 define cmd_check_data_rel
 	for obj in $(filter %.o,$^); do \
-		readelf -S $$obj | grep -qF .rel.local && { \
+		${CROSS_COMPILE}readelf -S $$obj | grep -qF .rel.local && { \
 			echo "error: $$obj has data relocations!" >&2; \
 			exit 1; \
 		} || true; \
